lib/electrumx_rpc/electrumx_client.dart

```diff
@@ -391,10 +392,14 @@ class ElectrumXClient {
 
       final List<dynamic> response;
       try {
-        response = jsonRpcResponse.data as List;
+        if (jsonRpcResponse.data is Map) {
```

Comment:
As this function is the batch request, the response should always be a list and never a map. Although there is a possibility that if there happens to be a single request in this batched call then it could be the case that a single response is then returned as a map rather than a map in a list of length 1. It may be worth adding an extra check to see of the `requestStrings` list is of length greater than 1 for a more detailed error message.

Reply:
more logging added in dc6d569433e80bf034de21a46b1bf08219981695